### ISSUE TITLE
NENGO Testing Tools

### ISSUE THREAD

**Issue by MatthewAKelly** [nengo/nengo#532]
It'd be nice if there was an easy way to test how well a NENGO model is approximating some function.

I'm dealing with a simple function that's being implemented by the connectivity from one ensemble to another, i.e.,

nengo.Connection(input_neurons, output_neurons, synapse=0.1, function=simple_function)

I'd like to know how good a job the model is doing approximating that function.

One way of doing this (that Terry suggested to me) is to create a copy of the model that turns the neurons off and just directly computes the function. The direct model could be run in real time alongside the neural model.

However, this is a recurrent model. The output is fed back in as input. Over time, the output of the direct and neural models will diverge. I'm not interested in that divergence over time, I'm only interested in the moment by moment error of the neural approximation.

In short, I'd like a tool that gives me (graphed perhaps) the moment by moment error of the neural approximation to some function.

**Comment by tbekolay** [comment]
This issue's a few years old now, so I'm going to close it. Hopefully @tcstewar's answer was satisfactory @MatthewAKelly! I've marked this issue with the `document this` label so that (eventually) we'll document the things that Terry mentioned better.
